src/localsocketcontroller.cpp

```diff
@@ -343,9 +351,17 @@ void LocalSocketController::parseCommand(const QByteArray& command) {
   logger.warning() << "Invalid command received:" << command;
 }
 
-void LocalSocketController::write(const QJsonObject& json) {
+void LocalSocketController::write(const QJsonObject& json,
+                                  const QString& expect) {
+  QByteArray payload = QJsonDocument(json).toJson(QJsonDocument::Compact);
+  payload.append('\n');
+
+  if (!expect.isEmpty()) {
+    m_messageExpected = expect;
```

Comment:
This m_messageExpected design assumes that the daemon will respond before the next write. Is that always true?

Also, does LocalSocketController::getBackendLogs need to pass an expected message?

Reply:
This assumes that the daemon will respond before the next write of another synchronous command (eg: `status` or `logs`). Asynchronous commands such as `connect` don't expect a response because the `Controller` handles that separately, or the daemon may not even respond at all if the tunnel is unable to establish a handshake.

In the event that such a race condition occurs, the last `write()` call with an `expected` value takes precedence when checking for responses, in which case we will still be able to detect a crashed daemon.

Reply:
`LocalSocketController::getBackendLogs()` is the only other case where we expect a synchronous response from the daemon, but it's not clear how long the daemon may take to generate that response if the log file is significant in size, so I decided to ignore the problem for the time being.

I suppose I could add another optional parameter to specify a timeout, in which case we could allow a little more time for the `getBackendLogs` command specifically.

Reply:
I have updated the response handling so that we can now handle multiple outstanding responses.